src/Console/Addon.php

```diff
@@ -0,0 +1,215 @@
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ */
+
+namespace Friendica\Console;
+
+use Console_Table;
+use Friendica\App;
+use Friendica\Content\Pager;
+use Friendica\Core\L10n;
+use Friendica\Core\Addon as AddonCore;
+use Friendica\Database\Database;
+use Friendica\Util\Strings;
+use RuntimeException;
+
+/**
+ * tool to manage addons on the current node
+ */
+class Addon extends \Asika\SimpleConsole\Console
+{
+	protected $helpOptions = ['h', 'help', '?'];
+
+	/**
+	 * @var App\Mode
+	 */
+	private $appMode;
+	/**
+	 * @var L10n
+	 */
+	private $l10n;
+	/**
+	 * @var Database
+	 */
+	private $dba;
+
+	protected function getHelp()
+	{
+		$help = <<<HELP
+console user - Modify addon settings per console commands.
+Usage
+	bin/console addon list all [-s|--start=0] [-c|--count=50] [-h|--help|-?] [-v]
+	bin/console addon list enabled [-s|--start=0] [-c|--count=50] [-h|--help|-?] [-v]
+	bin/console addon list disabled [-s|--start=0] [-c|--count=50] [-h|--help|-?] [-v]
+	bin/console addon enable <addonname> [-h|--help|-?] [-v]
+	bin/console addon disable <addonname> [-h|--help|-?] [-v]
+
+Description
+	Modify addon settings per console commands.
+
+Options
+    -h|--help|-? Show help information
+    -v           Show more debug information
+HELP;
+		return $help;
+	}
+
+	public function __construct(App\Mode $appMode, L10n $l10n, Database $dba, array $argv = null)
+	{
+		parent::__construct($argv);
+
+		$this->appMode     = $appMode;
+		$this->l10n        = $l10n;
+		$this->dba         = $dba;
+
+		AddonCore::loadAddons();
+	}
+
+	protected function doExecute()
+	{
+		if ($this->getOption('v')) {
+			$this->out('Class: ' . __CLASS__);
+			$this->out('Arguments: ' . var_export($this->args, true));
+			$this->out('Options: ' . var_export($this->options, true));
+		}
+
+		if (count($this->args) == 0) {
+			$this->out($this->getHelp());
+			return 0;
+		}
+
+		if ($this->appMode->isInstall()) {
+			throw new RuntimeException('Database isn\'t ready or populated yet');
+		}
+
+		$command = $this->getArgument(0);
+
+		switch ($command) {
+			case 'list':
+				return $this->list();
+			case 'enable':
+				return $this->enable();
+			case 'disable':
+				return $this->disable();
+			default:
+				throw new \Asika\SimpleConsole\CommandArgsException('Wrong command.');
+		}
+	}
+
+	/**
+	 * Lists plugins
+	 *
+	 * @return int Return code of this command
+	 *
+	 * @throws \Exception
+	 */
+	private function list()
+	{
+		$subCmd = $this->getArgument(1);
+		$start = $this->getOption(['s', 'start'], 0);
+		$count = $this->getOption(['c', 'count'], Pager::ITEMS_PER_PAGE);
```

Comment:
These parameters aren't used and should be removed from here and the list command synopses.